tiles/src/main/java/com/protomaps/basemap/layers/Places.java

```diff
@@ -17,10 +18,64 @@ public String name() {
     return "places";
   }
 
+  public void processNe(SourceFeature sf, FeatureCollector features) {
+    var sourceLayer = sf.getSourceLayer();
+    var kind = "";
+    var kind_detail = "";
+
+    var theme_min_zoom = 0;
+    var theme_max_zoom = 0;
+    if (sourceLayer.equals("ne_10m_populated_places")) {
+      theme_min_zoom = 1;
+      theme_max_zoom = 8;
+    }
+
+    // Test for props because of Natural Earth funk
+    if( sf.isPoint() && sf.hasTag("featurecla") && sf.hasTag("min_zoom" ) ) {
+      switch (sf.getString("featurecla")) {
+        case "Admin-0 capital":
+        case "Admin-0 capital alt":
+        case "Admin-0 region capital":
+          kind = "city";
+          break;
+        case "Admin-1 capital":
+        case "Admin-1 region capital":
+          kind = "city";
+          break;
+        case "Populated place":
+          kind = "city";
+          break;
+        case "Historic place":
+          kind = "locality";
+          kind_detail = "hamlet";
+          break;
+        case "Scientific station":
+          kind = "locality";
+          kind_detail = "scientific_station";
+          break;
+      }
+    }
+
+    if( kind != "" ) {
+      var feat = features.point(this.name())
+              .setAttr("name", sf.getString("name"))
+              .setAttr("pmap:min_zoom", sf.getLong("min_zoom"))
+              .setZoomRange(sf.getString("min_zoom") == null ? theme_min_zoom : (int)Double.parseDouble(sf.getString("min_zoom")), theme_max_zoom)
+              .setAttr("pmap:kind", kind)
+              .setAttr("pmap:kind_detail", kind_detail)
+              .setAttr("population", sf.getString("pop_max"))
+              .setAttr("population_rank", sf.getString("rank_max"))
+              .setAttr("wikidata_id", sf.getString("wikidata"))
+              .setBufferPixels(128);
+
+      NeNames.setNeNames(feat, sf, 0);
+    }
+  }
   @Override
   public void processFeature(SourceFeature sf, FeatureCollector features) {
     if (sf.isPoint() &&
-      (sf.hasTag("place", "suburb", "town", "village", "neighbourhood", "city", "country", "state"))) {
+      (sf.hasTag("place", "suburb", "town", "village", "neighbourhood", "city", "country", "state", "province"))) {
+      Integer population = sf.getString("population") == null ? 0 : (int)Double.parseDouble(sf.getString("population"));
```

Comment:
We always want to have population values, default to 0 if missing and backfill by place type down below.